ui.tests/test-module/specs/contentfragment/contentfragment.authoring.spec.js

```diff
@@ -0,0 +1,143 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+/// <reference types="cypress" />
+
+// This recipe is to test the Content Fragment Component functionality
+
+// We are going to test a few things:
+// 1. Open Edit layer
+// 2. Perform few operations
+
+// Be sure to run the aem server
+// before running the tests below.
+
+const commons = require('../../libs/commons/commons'),
+    sitesSelectors = require('../../libs/commons/sitesSelectors'),
+    afConstants = require('../../libs/commons/formsConstants');
+const siteSelectors = require("../../libs/commons/sitesSelectors");
+
+// todo: beta specific form authoring test cases are not run (only common functionality [of beta and to be GA] test are executed)
+describe('Content Fragment - Authoring', function () {
+
+    const dropContentFragmentInContainer = function() {
+        const dataPath = "/content/forms/af/core-components-it/blank/jcr:content/guideContainer/*",
+            responsiveGridDropZoneSelector = sitesSelectors.overlays.overlay.component + "[data-path='" + dataPath + "']";
+        cy.selectLayer("Edit");
+        cy.insertComponent(responsiveGridDropZoneSelector, "Content Fragment", afConstants.components.resourceType.contentfragment);
+        cy.get('body').click( 0,0);
+    }
+
+    const dropContentFragmentInSites = function() {
+        const dataPath = "/content/core-components-examples/library/adaptive-form/container/jcr:content/root/responsivegrid/demo/component/formContainer/*",
+            responsiveGridDropZoneSelector = sitesSelectors.overlays.overlay.component + "[data-path='" + dataPath + "']";
+        cy.selectLayer("Edit");
+        cy.insertComponent(responsiveGridDropZoneSelector, "Content Fragment", afConstants.components.resourceType.contentfragment);
+        cy.get('body').click( 0,0);
+    }
+
+    const testContentFragmentBehaviour = function(cfEditPathSelector, cfDrop, isSites) {
+        if (isSites) {
+            dropContentFragmentInSites();
+        } else {
+            dropContentFragmentInContainer();
+        }
+        cy.openEditableToolbar(sitesSelectors.overlays.overlay.component + cfEditPathSelector);
+        cy.invokeEditableAction("[data-action='CONFIGURE']");
+        // Check If Dialog Options Are Visible
+        cy.get("[name='./fragmentPath']")
+            .should("exist");
+        cy.get("[name='./displayMode']")
+            .should("exist");
+
+        cy.get("[name='./fragmentPath'] input[role='combobox']").should("exist").clear()
+            .type("/content/dam/wknd/library/sample-assets/adobe-headquarters", { delay: 0 }).click().then(x => {
```

Comment:
do we need the delay:0 option here even after adding the then statement

Reply:
delay: 0 types the text at one go, instead of taking some time to actually type. So adding it will not affect the test case timings.